lib/iaas/AzureClient.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+const _ = require('lodash');
+const Promise = require('bluebird');
+const azureStorage = require('azure-storage');
+const logger = require('../logger');
+const errors = require('../errors');
+const utils = require('../utils');
+const ComputeClient = require('./ComputeClient');
+const CONST = require('../constants');
+const BaseCloudClient = require('./BaseCloudClient');
+const NotFound = errors.NotFound;
+const Unauthorized = errors.Unauthorized;
+const Forbidden = errors.Forbidden;
+
+class AzureClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    this.storage = this.constructor.createStorageClient(_
+      .chain(this.settings)
+      .omit('name')
+      .set('provider', this.provider)
+      .value()
+    );
+    this.computeClient = this.constructor.createComputeClient(_
+      .chain(this.settings)
+      .set('provider', this.provider)
+      .value()
+    );
+    Promise.promisifyAll(this.computeClient.snapshots);
+
+    this.storage.on('log::*', function onmessage(message, obj) {
+      const event = _.nth(this.event.split('::'), 1);
+      const level = _.includes([
+        'warn',
+        'info',
+        'verbose',
+        'debug'
+      ], event) ? event : 'debug';
+      if (obj) {
+        logger.log(level, message, obj);
+      } else {
+        logger.log(level, message);
+      }
+    });
+  }
+
+  getContainer(container) {
```

Comment:
rename variable from `container` to `containerName`

Reply:
If you feel it is not a must required.... then can we keep the name intact. In cloudproviderclient class also we are using same name.